staging/src/k8s.io/apimachinery/pkg/api/validate/schema.go

```diff
@@ -72,3 +72,14 @@ func Optional[T comparable](_ operation.Context, fldPath *field.Path, value, _ *
 	}
 	return field.ErrorList{field.Required(fldPath, "optional value was not specified")}
 }
+
+// MaxItems verifies that the specified slice is not longer than max items.
```

Comment:
should it apply to maps, too?  We do have a few places that do maps, and DRA introduces a couple more.

If so, when emitting code we need to call into `MaxItemsSlice` vs `MaxItemsMap` or something

Reply:
Thanks for reviewing, @thockin . The "maxItems" here matches the openapi(json) schema which doesn't apply to maps. How about we having an additional marker `+maxItemsMap` dedicated for the maps?

Reply:
I see.  OK, if there's already a standard name for that, then I am OK.  "properties" is a weird JSON-ish term, but if it is even a little standard, I guess that's fine :)